src/content/docs/en/guides/cms/datocms.mdx

```diff
@@ -0,0 +1,202 @@
+type: cms
+stub: false
+service: DatoCMS
+i18nReady: true
+---
+import FileTree from '~/components/FileTree.astro'
+
+[DatoCMS](https://datocms.com/) is the headless CMS for the modern web.
+
+## Integrating with Astro
+
+In this guide, you will learn how to get the data of a single page from DatoCMS using Astro.
+
+## Prerequisites
+
+To get started, you will need to have the following:
+
+- **A DatoCMS Account and project** - if you don't have an account, you can [sign up for a free account](https://dashboard.datocms.com/signup)
+- **The read-only API Key for your DatoCMS Project** - you can find it in the admin dashboard of your Project, under "Settings" > "API Tokens"
+
+## Setting up the credentials
+
+For safety, it is best practice to save your DatoCMS API key in an **environment variable**.
+Let's start by creating a new file named `.env` in the root of your Astro project. In there, you can create a new variable for our DatoCMS API Key as follows: 
+
+```ini title=".env"
+DATOCMS_API_KEY=YOUR_API_KEY
+```
+
+If you're using TypeScript for this project, you can declare the typing of our environment variable in the `env.d.ts` file in the `src` folder. If you didn't opt-in for TypeScript usage when setting up your project, you can create the file yourself.
+
+```ts title="src/env.d.ts"
+interface ImportMetaEnv {
+  readonly DATOCMS_API_KEY: string;
+}
+```
+
+:::tip
+Read more about [using environment variables](/en/guides/environment-variables/) and `.env` files in Astro.
+:::
+
+Your root directory should now include these new files:
+
+<FileTree title="Project Structure">
+- src/
+  - **env.d.ts**
+- **.env**
+- astro.config.mjs
+- package.json
+</FileTree>
+
+## Create a Model in DatoCMS
+
+Before you can continue, you'll need to create a homepage in DatoCMS. Go to the admin dashboard of your Project, and navigate to the "Settings" tab. Click on "Models", and create a new Model called "Home". Make sure to toggle the "Single instance" toggle. In the model, add a new text field for a title.
+
+Navigate to the "Content" tab and click on your newly created Homepage. You can now add a title. Save the page, and continue.
+
+## Fetching data
+
+Navigate to `index.astro`. Fetch the content for `home` using the GraphQL API.
+
+```astro title="src/pages/index.astro"
+---
+const response = await fetch('https://graphql.datocms.com/', {
+	method: 'POST',
+	headers: {
+		'Content-Type': 'application/json',
+		Accept: 'application/json',
+		Authorization: `Bearer ${import.meta.env.DATOCMS_API_KEY}`,
+	},
+	body: JSON.stringify({
+		query: `query Homepage {
+          home {
+            title
+          }
+        }
+      `,
+	}),
+});
+
+const json = await response.json();
+const data = json.data.home;
+---
+
+<h1>{data.title}</h1>
+
+```
+
+This GraphQL query will fetch the "title" field in the "home" page from your DatoCMS Project. When you refresh your browser, you should see the title on your page.
+
+## Dynamically fetch blocks
```

Comment:
I'm going to stop this review and go into the file to make some edits directly to this section, not because it's terrible! :smile: But, because I really want to reorder what's here, and the only way to do that in this interface is one 80-line block revised.

I'll release all these review comments for the smaller things that can be fixed in place. Then separately, I'll do some reordering/reworking of the Dynamically fetching blocks" section. In particular, I think it's important to show building the `.astro` components to correspond to the modular content blocks first, before the page that tries to render components that don't exist yet! I see a natural flow as:

- In addition to full page content, you can create modular content blocks in DatoCMS (link)
- Each modular content block you want to represent on your site should have a corresponding `.astro` component
- Here is an example `<Text />` Astro component for multi-paragraph text
- Here's an updated GraphQL query that fetches this modular content block, and renders it via the Astro component

I would also remove the unnecessarily complicated switch case, and just display the title from before, then below it, this text content.

So, that change will be incoming, and then of course we can make changes to *it* once it's in place here!